bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/core/mapping/operation/initializer/PropertyInitializerFactory.java

```diff
@@ -31,19 +38,50 @@ public IPropertyInitializer newPropertyInitializer(final FieldToContractInputMap
             return new SimpleFieldPropertyInitializer(firstMultipleParentBusinessObject(mapping), (SimpleField) field, mapping.getContractInput());
         } else if (field instanceof RelationField) {
             final RelationField relationField = (RelationField) field;
+            final Type relationType = relationField.getType();
             if (Objects.equals(field, rootField)) {
+                if (relationType == Type.COMPOSITION) {
+                    return field.isCollection() ?
+                            new BusinessObjectListInitializer(relationField, mapping.getContractInput(), toRefName(mapping, data)) :
```

Comment:
in case of collection the "then" is similar for composition and aggregation, we can factorize